hera_cal/nucal.py

```diff
@@ -1013,3 +1032,516 @@ def project_u_model_comps_on_spec_axis(u_model_comps, spectral_filters):
+        idealized_antpos = abscal._get_idealized_antpos(cal_flags, self.antpos, data.pols())
+
+        if estimate_degeneracies:
+            model = evaluate_foreground_model(
+                self.radial_reds, init_model_comps, spatial_filters=self.spatial_filters, spectral_filters=self.spectral_filters
+            )
+            amplitude, tip_tilt = self._estimate_degeneracies(data, model, data_wgts)
+        else:
+            amplitude = {pol: np.ones((data.shape)) for pol in data.pols()}
+            tip_tilt = {
+                pol: np.zeros((idealized_antpos[list(idealized_antpos.keys())[0]].shape[0], data.shape[0], data.shape[1])) 
+                for pol in data.pols()
+            }
+        
+        # Initialize model parameters and metadata dictionaries for storing results
+        model_parameters = {}
+        metadata = {}
+
+        for pol in data.pols():
+            # Initialize model parameters
+            init_model_parameters = {}
+
+            # Separate data into real and imaginary components
+            data_real = jnp.array([
+                data[blkey].real for rdgrp in self.radial_reds.get_pol(pol) for blkey in rdgrp
+            ])
+            data_imag = jnp.array([
+                data[blkey].imag for rdgrp in self.radial_reds.get_pol(pol) for blkey in rdgrp
+            ])
+
+            # Initialize model parameters
+            init_model_parameters = {
+                "fg_r": [init_model_comps[rdgrp[0]].real for rdgrp in self.radial_reds.get_pol(pol)],
+                "fg_i": [init_model_comps[rdgrp[0]].imag for rdgrp in self.radial_reds.get_pol(pol)],
+                "amplitude": amplitude[pol],
+                "tip_tilt": tip_tilt[pol]
+            }
+            # unpack wgts/filters
+            wgts = jnp.array([
+                data_wgts[blkey] for rdgrp in self.radial_reds.get_pol(pol) for blkey in rdgrp
+            ])
+
+            # Set spectral filters 
+            spatial_filters = [
+                jnp.array([self.spatial_filters[blkey] for blkey in rdgrp]) 
+                for rdgrp in self.radial_reds.get_pol(pol)
+            ]
+
+            # Compute idealized baseline vectors
+            idealized_blvecs = np.array([
+                idealized_antpos[blkey[1]] - idealized_antpos[blkey[0]]
+                for rdgrp in self.radial_reds.get_pol(pol) for blkey in rdgrp
+            ])
+
+            # Run optimization
+            _model_parameters, _metadata = _nucal_post_redcal(
+                data_real, data_imag, wgts, init_model_parameters, optimizer, spectral_filters=self.spectral_filters, 
+                spatial_filters=spatial_filters, idealized_blvecs=idealized_blvecs, maxiter=maxiter, convergence_criteria=convergence_criteria,
+                minor_cycle_maxiter=minor_cycle_maxiter
+            )
+
+            # Pack model parameters and metadata
+            metadata[pol] = _metadata
+            model_parameters[pol] = _model_parameters
```

Comment:
I think this can be avoided by just doing `metadata[pol], model_parameters[pol] = _nucal_post_redcal(`

Reply:
Good point, removed those lines